### PR TITLE
Fixed a bug in SearchSource.prototype._updateStore. Resolves #7, #15

### LINKED ISSUE #7: Only getting results for every second character
I'm getting search-results just every second character I type. It seems to be a problem with rendering because on the server I get results on every character.

### PR DESCRIPTION
**Pull request by woniesong92**
The problem was `_.difference` was not working correctly to compare `id` objects. I fixed the issue by comparing `id strings` instead when deciding which items to remove from the client database.

### CODE REVIEW

**Comment by woniesong92** [comment]
this fixes the bug described in #7 and #15

File: lib/client.js [hunk 1]
```diff
@@ -79,8 +79,14 @@ SearchSource.prototype._updateStore = function(data) {
     self.store.update(item._id, item, {upsert: true});
   });
 
-  var removedItem = _.difference(storeIds, currentIds);
-  removedItem.forEach(function(id) {
+  // Remove items in client DB that we no longer need
+  var currentIdStrs = _.map(currentIds, function(currentId){ 
+    return currentId._str;
```
Comment (arunoda) [inline lib/client.js]: It's possible `currentId` to be just string. So, we need to check that as well.
May be try to use the `typeof` check.

File: lib/client.js [hunk 2]
```diff
@@ -79,8 +79,14 @@ SearchSource.prototype._updateStore = function(data) {
...
   });
 
-  var removedItem = _.difference(storeIds, currentIds);
-  removedItem.forEach(function(id) {
+  // Remove items in client DB that we no longer need
+  var currentIdStrs = _.map(currentIds, function(currentId){ 
+    return currentId._str; 
+  });
+  var removedItems = _.reject(storeIds, function(storeId) {
+    return _.contains(currentIdStrs, storeId._str);
```
Comment (arunoda) [inline lib/client.js]: Same as the above comment for storeId

**Comment by woniesong92** [comment]
done

File: lib/client.js [hunk 3]
```diff
@@ -79,8 +79,15 @@ SearchSource.prototype._updateStore = function(data) {
     self.store.update(item._id, item, {upsert: true});
   });
 
-  var removedItem = _.difference(storeIds, currentIds);
-  removedItem.forEach(function(id) {
+  // Remove items in client DB that we no longer need
+  var currentIdStrs = _.map(currentIds, function(currentId){ 
+    return (typeof(currentId) == 'string') ? currentId : currentId._str;
```
Comment (arunoda) [inline lib/client.js]: Use `===` instead.

**Comment by woniesong92** [comment]
thanks

**Comment by woniesong92** [comment]
Can this be merged?

File: .npm/package/.gitignore [hunk 4]
```diff
@@ -0,0 +1 @@
+node_modules
```
Comment (arunoda) [inline .npm/package/.gitignore]: remove .npm directory from git.

File: package.js [hunk 5]
```diff
@@ -2,7 +2,7 @@ Package.describe({
   "summary": "Reactive Data Source for Search",
   "version": "1.3.0",
   "git": "https://github.com/meteorhacks/search-source.git",
-  "name": "meteorhacks:search-source"
+  "name": "mrt:search-source"
```
Comment (arunoda) [inline package.js]: undo this change.
It should be `meteorhacks:search-source`

**Comment by woniesong92** [comment]
Last changes made. Can this be merged so other people can use this package?

**Comment by arunoda** [comment]
Thanks.
Merged with some changes.
Released as: `1.3.1`

**Pull request merged**